pkg/operation/controlplanesecrets/secrets_deployer.go

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplanesecrets
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	gardencorev1alpha1helper "github.com/gardener/gardener/pkg/apis/core/v1alpha1/helper"
+	"github.com/gardener/gardener/pkg/utils/infodata"
+	"github.com/gardener/gardener/pkg/utils/secrets"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SecretsDeployer defines methods which can be used to deploy Secrets. The configuration from these secrets is taken from the passed in
+// GardenerResourceDataList. The deployed secrets are saved in the DeployedSecrets.
+type SecretsDeployer struct {
+	*SecretsManager
+	APIServerBasicAuth *secrets.BasicAuth
+	StaticToken        *secrets.StaticToken
+
+	ExistingSecrets map[string]*corev1.Secret
+	DeployedSecrets map[string]*corev1.Secret
+
+	mutex *sync.RWMutex
+}
+
+// NewSecretsDeployer creates a new SecretsDeployer.
+func NewSecretsDeployer(
```

Comment:
Does it make sense to only pass the `SecretsManager` object instead of all its related attributes?